File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2021s/Id06TC2/Strategy/Command/Sportsmen.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Command
{
    public class Sportsmen
    {
        private string name;
        private StrategyRun run;
        public Sportsmen(string name,StrategyRun run) 
        {
            this.name=name; 
            this.run=run;
        }
        public string Run() { return run.execute(); }
        public string Jump() { return "Jumping"; }
        public string Swim() { return "Swimming"; }
        public string execute(SportsmenCommand cmd)
        {
            return name + ": " + cmd.execute(this);
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2021s/IZ08IO1/Command/src/Sportsmen.java
public class Sportsmen
{
   private String name;
   private SportsmenStrategyRun strategyrun;
   public Sportsmen(String name,SportsmenStrategyRun strategyrun)
   {
      this.name=name;
      this.strategyrun=strategyrun;
   }
   public String run() { return strategyrun.execute(); }
   public String jump() { return "Jumping"; }
   public String swim() { return "Swimming"; }
   public String execute(SportsmenCommand cmd) { return name+": "+cmd.execute(this); }
}
